cookbooks/creating_a_behat_extension.rst

```diff
@@ -0,0 +1,285 @@
+Creating a Behat extension
+==========================
+
+Extensions are particularly useful when configuration becomes a necessity.
+
+In this cookbook, we will create a simple extension named ``LogsExtension`` that logs scenario durations and cover:
+
+#. Setting up the context
+#. Creating the extension
+#. Initializing the context
+
+Setting Up the Context
+----------------------
+
+First, we need to create a class that will handle the logging. There are several ways to do this (you can hook
+directly into behat's event dispatcher, for example).
+
+For our case, we will create a ``Context`` and use the :doc:`normal tagged hooks</user_guide/context/hooks>`
+that you might have used in your own contexts.
+
+This is a straightforward approach, and means we can also show how to initialise any context with custom
+configuration or dependencies.
+The ``LogsExtension`` will provide a ``LogsContext`` that hooks into scenarios to log their start and end times.
+
+Directory structure:
+
+.. code-block::
+  
+  src/
+      Context/
+          LogsContext.php   # This is where we'll implement our logging logic
+
+
+The code for ``LogsContext.php``:
+
+.. code-block:: php
+
+  <?php
+  
+  namespace Behat\LogsExtension\Context;
```

Comment:
I suggest using a different namespace that `Behat`, to avoid suggesting that extensions should be using the `Behat` namespace. The `Behat` namespace should be used only by official extensions. Third-party extensions should be using the top-level namespace of that third-party instead.

Reply:
Surely a typo, it's not in my own [boilerplate](https://github.com/monitaurus/behat-extension-boilerplate) 🤷 

It's fixed thanks: 8e8526805ae58c80b79ef17954b1c7dd6a9588be